chalice/local.py

```diff
@@ -187,24 +187,43 @@ def do_OPTIONS(self):
         else:
             # Otherwise this is a preflight request which we automatically
             # generate.
-            self._send_autogen_options_response()
+            self._send_autogen_options_response(lambda_event)
 
     def _has_user_defined_options_method(self, lambda_event):
         # type: (EventType) -> bool
         route_key = lambda_event['requestContext']['resourcePath']
         return 'OPTIONS' in self.app_object.routes[route_key]
 
-    def _send_autogen_options_response(self):
-        # type:() -> None
+    def _send_autogen_options_response(self, lambda_event):
+        # type:(EventType) -> None
+        route_key = lambda_event['requestContext']['resourcePath']
+        route_dict = self.app_object.routes[route_key]
+        route_methods = list(route_dict.keys())
+
+        # Chalice ensures that routes with multiple views have the same
+        # CORS configuration, so if any view has a CORS Config we can use
+        # that config since they will all be the same.
+        cors_config = route_dict[route_methods[0]].cors
+        cors_headers = cors_config.get_access_control_headers()
+
+        # The keys will be ordered in python 3 but in python 2 they will not
+        # be. To assert this header is sent correctly we need to introduce
```

Comment:
It is not necessarily true about the need to sort this for the assertion. You could always just split the header value by commas and assert that the sorted list is the same, but that is a bit more tedious. I only bring it up because it is a little weird to have this sorted in the core logic when it is only being sorted for the test. I'm about +0 in updating this especially since we do the same logic in the non chalice local tests (i.e. assert the header values in the a split list).